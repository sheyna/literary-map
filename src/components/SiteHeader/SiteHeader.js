import {
  Link,
} from "react-router-dom";
import LoginButton from "../LoginButton/LoginButton";
import LogoutButton from "../LogoutButton/LogoutButton";
import { useAuth0 } from "@auth0/auth0-react";
import styles from './SiteHeader.module.css';

function SiteHeader({ showDecorativeTitle }) {
	const { isAuthenticated, isLoading } = useAuth0();

	return (
		<header className={styles.SiteHeader}>
			<menu>
				<h2><Link to="/" style={{ textDecoration: 'none' }}>LIT</Link></h2>
				<nav>
					<ul>
						<li><Link to="/" style={{ textDecoration: 'none' }}>Home</Link></li>
						<li><Link to="/about" style={{ textDecoration: 'none' }}>About</Link></li>
						<li><Link to="/list" style={{ textDecoration: 'none' }}>List</Link></li>
					</ul>
					{
						isAuthenticated 
							? <LogoutButton/> 
							: <LoginButton/>
					}
				</nav>
			</menu>
			{ 
				showDecorativeTitle 
					&& <div className={styles.title}>
						<h1>Literary Map</h1>
						<h1>of England</h1>
					</div>
			}
		</header>
	)
}

export default SiteHeader;
