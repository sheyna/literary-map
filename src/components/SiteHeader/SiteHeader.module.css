.SiteHeader {
  position: relative;
  top: 0;
  right: 0;
  z-index: 20;
  width: 100%;
  background-color: #FEFAF5;
  height: 6rem;
}

.title {
  position: relative;
  padding-top: 1rem;
  width: 100%;
  z-index: 90;
}

.SiteHeader menu {
  margin: 0;
  padding-inline-start: 1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  z-index: 100;
  position: relative;
  height: 2rem;
}

.SiteHeader menu li > a {
  text-decoration: none;
  padding: .5rem;
  border-bottom: 1rem solid transparent;
}

.SiteHeader menu a:hover {
  color: #bd4089;
  cursor: pointer;
  border-color: #bd4089;
}

.SiteHeader h1 {
  font-family: 'Satisfy', cursive, serif;
  color: #bd4089;
  font-size: 2.5rem;
  line-height: 85%;
  text-align: center;
  margin: 0;
  position: relative;
  transform: rotate(-15deg)
}

.SiteHeader h2 a,
.SiteHeader h2 {
  font-family: 'Open Sans', sans-serif;
  font-weight: 800;
  font-size: 2rem;
  text-decoration: none;
  color: #bd4089;
  margin: .5rem 0 0;
  padding: .5rem;
  display: inline-block;
}

.SiteHeader ul {
  list-style-type: none;
  padding: 0;
  margin: 0 1rem 0 0;
  display: flex;
  font-size: 1rem;
}
.SiteHeader ul li {
  padding: 0 0.4rem;
}

@media screen and (max-width: 759px) {
  .SiteHeader h1:first-of-type {
    margin: 1rem auto .3rem;
  }
  .SiteHeader h1:nth-of-type(2) {
    margin: 0 auto 0;
    padding-left: 5rem;
  }
}

@media screen and (min-width: 380px) {
  .SiteHeader h1 {
    font-size: 3.5rem;
    width: auto;
  }
}

@media screen and (min-width: 760px) {
  .SiteHeader h1 {
    font-size: 8vw;
    margin-left: 48%;
  }
  .SiteHeader {
    height: 3rem;
  }
  .SiteHeader h1:first-of-type {
    margin-top: 3rem;
    margin-bottom: .8rem
  }
  .SiteHeader h1:nth-of-type(2) {
    padding-left: 12vw;
  }
}
@media screen and (min-width: 1025px) {
  .SiteHeader h1 {
    font-size: 10vw;
    margin-left: 44%;
  }
  .SiteHeader h1:first-of-type {
    margin-top: 5rem;
  }
}

@media screen and (min-width: 1450px) {
  .SiteHeader h1 {
    font-size: 12vw;
    margin-left: 38%;
  }
}
