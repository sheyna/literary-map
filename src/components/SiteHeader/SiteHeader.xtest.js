import ReactTestRenderer from 'react-test-renderer';
import SiteHeader from  './SiteHeader';

// TODO, update this test
describe('SiteHeader component', () => {
  it('should match snapshot', () => {
    const component = ReactTestRenderer.create(
      <SiteHeader 
        showDecorativeTitle={true}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
