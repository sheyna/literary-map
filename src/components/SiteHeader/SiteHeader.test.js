import ReactTestRenderer from 'react-test-renderer';
import SiteHeader from  './SiteHeader';

describe('SiteHeader component', () => {
  it('should match snapshot', () => {
    const component = ReactTestRenderer.create(
      <SiteHeader />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});