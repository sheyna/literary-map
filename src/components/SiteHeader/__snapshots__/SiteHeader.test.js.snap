// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteHeader component should match snapshot 1`] = `
<header
  className="SiteHeader"
>
  <div
    className="title"
  >
    <h1>
      Literary Map
    </h1>
    <h1>
      of England
    </h1>
  </div>
</header>
`;
